models/app.py

```diff
@@ -0,0 +1,119 @@
+import datetime
+import json
+import logging
+import os
+from datetime import datetime
+
+from db_models import Activity
+from flask import Flask, jsonify, request
+from flask_mongoengine import MongoEngine
+from mongoengine.queryset.visitor import Q
+
+DATE_FMT = '%Y-%m-%d %H:%M:%S'
+
+log = logging.getLogger()
+logging.basicConfig(filename="models/mongo.log", filemode='a',
+                    format='%(asctime)s, %(levelname)s %(message)s',
+                    datefmt='%H:%M:%S',
+                    level=logging.DEBUG)
+
+app = Flask(__name__)
+
+
+@app.route('/')
+def index():
+    return jsonify(status=200, message='MongoDB app!', data=[])
+
+
+@app.route('/activity', methods=['POST'])
+def add_activity():
+    data_dict = request.get_json(force=True)
+    if not isinstance(data_dict, dict):
+        return jsonify(status=400, message="Invalid payload data. Data "
+                                           "should be single JSON", data=[])
+    _items = list()
+    for item in data_dict['items']:
+        _new_activity = Activity()
+        _new_activity = _new_activity.from_json(json.dumps(item))
+        try:
+            _new_activity.validate()
+        except Exception as e:
+            log.error(e.to_dict())
+            return jsonify(status=400, message=e.to_dict(), data=[])
+        _items.append(_new_activity)
+    Activity.objects.insert(_items)
+    msg = "Total {} activities added successfully.".format(len(_items))
+    log.info(msg)
+    return jsonify(status=200, message=msg, data=[])
+
+
+@app.route('/activity/<int:ph_no>')
+def get_activity(ph_no):
```

Comment:
![image](https://user-images.githubusercontent.com/20097156/74813775-b6063700-52aa-11ea-8e7e-72a24b05ddeb.png)

phone number is not in the DB so shouldn't it return some warning or 4xx error?

Reply:
I'm not registering a user. We do not have separate Collections for Users.

Reply:
What do you mean? I was referring to the function itself and validating API call.
Eg, phone number 1234567 is not in the db then request for "$host/activity/1234567" should be denied but the function returns status 200 with empty data. When valid `ph_no`  is not given then, should the API call be allowed?